Added new Model provider: Straico

niteshmanav opened the pull request:
Documentation: https://documenter.getpostman.com/view/5900072/2s9YyzddrR

niteshmanav committed:
Create Straico provider.toml

niteshmanav committed:
Create anthropic_claude-3-5-haiku-20241022.toml

niteshmanav committed:
Create anthropic_claude-3-7-sonnet.toml

niteshmanav committed:
Create deepseek_deepseek-r1-nitro.toml

niteshmanav committed:
Create o3-mini.toml

niteshmanav committed:
Create anthropic_claude-sonnet-4.toml

niteshmanav committed:
Create x-ai_grok-4.toml

niteshmanav committed:
Update provider.toml

fwang commented:
Thanks @niteshmanav! Some info of the provider and models are missing, causing the test to fail.

niteshmanav committed:
Update provider.toml

added npm: straico-api

niteshmanav committed:
Update provider.toml

updated npm - @ai-sdk/straico-api

niteshmanav committed:
Merge branch 'sst:dev' into dev

niteshmanav commented:
Thank you @fwang for looking into this.

I’m not a coder by trade but I’m enthusiastic about OpenCode and wanted to add support for Straico (a paid service similar to OpenRouter). Using Straico’s API docs (https://documenter.getpostman.com/view/5900072/2s9YyzddrR), the models.dev README (https://github.com/sst/models.dev/blob/dev/README.md), and some help from AI, I implemented the integration.

I discovered an issue caused by an empty npm dependency, so I added "@ai-sdk/straico-api" to package. That should resolve the problem, but if it still fails to run, I’d appreciate your help troubleshooting.

niteshmanav committed:
Update provider.toml

Added api

niteshmanav reviewed (commented):
Just added 
api = "https://api.straico.com/v1"
npm = "@ai-sdk/openai-compatible"

niteshmanav commented:
Hey @fwang Could you please look into this once? Thank you.

fwang commented:
Thanks @niteshmanav. Straico seems to be charging by coin/word. The pricing structure is quite different as the standard $/token. And I wasn't able to find any info on what is considered a word. Making it tricky to merge.
